src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -239,463 +241,61 @@ private Object createObject(String className) throws CheckstyleException {
+        Map<String, String> returnValue;
+        try {
+            returnValue = ClassPath.from(loader).getTopLevelClassesRecursive(BASE_PACKAGE).stream()
+                    .map(ClassPath.ClassInfo::load)
+                    .filter(PackageObjectFactory::shouldAddToMap)
+                    .collect(Collectors.toMap(Class::getSimpleName, Class::getCanonicalName));
+        }
+        catch (IOException ignore) {
+            returnValue = new HashMap<>();
+        }
+        return returnValue;
     }
 
     /**
-     * Fill short-to-full module names map with modules from filefilters package.
+     * Whether the class should be added to the map.
+     * @param clazz the class to check
+     * @return true if the class should be added to the map
      */
-    private static void fillModulesFromFilefiltersPackage() {
-        NAME_TO_FULL_MODULE_NAME.put("BeforeExecutionExclusionFileFilter",
-                BASE_PACKAGE + ".filefilters.BeforeExecutionExclusionFileFilter");
+    private static boolean shouldAddToMap(Class<?> clazz) {
+        return !Modifier.isAbstract(clazz.getModifiers())
+                && !clazz.getSimpleName().startsWith("Input")
+                && isNameCorrespondingToCorrectPackage(
+                        clazz.getSimpleName(), clazz.getPackage().getName());
     }
 
     /**
-     * Fill short-to-full module names map with modules from filters package.
+     * Whether the class name is corresponding to a correct package.
+     * @param className the class name to check.
+     * @param packageName the package of the class.
+     * @return true if the class the class name is corresponding to a correct package
      */
-    private static void fillModulesFromFiltersPackage() {
-        NAME_TO_FULL_MODULE_NAME.put("CsvFilter",
-                BASE_PACKAGE + ".filters.CsvFilter");
-        NAME_TO_FULL_MODULE_NAME.put("IntMatchFilter",
-                BASE_PACKAGE + ".filters.IntMatchFilter");
-        NAME_TO_FULL_MODULE_NAME.put("IntRangeFilter",
-                BASE_PACKAGE + ".filters.IntRangeFilter");
-        NAME_TO_FULL_MODULE_NAME.put("SeverityMatchFilter",
-                BASE_PACKAGE + ".filters.SeverityMatchFilter");
-        NAME_TO_FULL_MODULE_NAME.put("SuppressionCommentFilter",
-                BASE_PACKAGE + ".filters.SuppressionCommentFilter");
-        NAME_TO_FULL_MODULE_NAME.put("SuppressionFilter",
-                BASE_PACKAGE + ".filters.SuppressionFilter");
-        NAME_TO_FULL_MODULE_NAME.put("SuppressWarningsFilter",
-                BASE_PACKAGE + ".filters.SuppressWarningsFilter");
-        NAME_TO_FULL_MODULE_NAME.put("SuppressWithNearbyCommentFilter",
-                BASE_PACKAGE + ".filters.SuppressWithNearbyCommentFilter");
+    private static boolean isNameCorrespondingToCorrectPackage(
+            String className, String packageName) {
+        return packageName.contains(BASE_PACKAGE + ".checks") && className.endsWith(CHECK_SUFFIX)
```

Comment:
See my comments at https://github.com/checkstyle/checkstyle/issues/3607#issuecomment-291342875

> > our naming notation that Check class should be "*Check".

> We dropped this convention in test's CheckUtil because it missed out on our own checks that end with Holder instead of Check.
I think checking the class' inheritance is enough, who cares what it is named especially for 3rd party checks.

Please change this to do a check on class inheritance instead.
Since this is for 3rd party checks, we can't assume package or class name will make sense.

Please look at test code which does similar at
https://github.com/checkstyle/checkstyle/blob/ed5d4d98da5963b658eafaaa3aaee98ba989db65/src/test/java/com/puppycrawl/tools/checkstyle/internal/CheckUtil.java#L146
https://github.com/checkstyle/checkstyle/blob/ed5d4d98da5963b658eafaaa3aaee98ba989db65/src/test/java/com/puppycrawl/tools/checkstyle/internal/CheckUtil.java#L199

Maybe we may want to consider moving those utility methods to production, then along with your changes, all 3rd party libraries will be able to recognize and identify our modules by type atleast.